tests/testthat/test-get_delay_estimate.R

```diff
@@ -1,124 +1,161 @@
+  full_triangle <- lapply(complete_counts, function(x) round(x * complete_pmf))
+  full_triangle <- do.call(rbind, full_triangle)
 
-  # Test 9: Test that you get the correct delay with a complete triangle
-  complete_triangle <- matrix(
-    c(
-      80, 50, 25, 10,
-      100, 50, 30, 20,
-      90, 45, 25, 20,
-      80, 40, 15, 5,
-      70, 30, 10, 10
-    ),
-    nrow = 5,
-    byrow = TRUE
-  )
   delay_pmf <- get_delay_estimate(
-    reporting_triangle = complete_triangle,
+    reporting_triangle = full_triangle,
     max_delay = 3,
     n = 5
   )
-  pmf <- colSums(complete_triangle) / sum(complete_triangle)
-  expect_identical(delay_pmf, pmf)
-
-  # Test 10: Test that you get a warning with zeros in the bottom right
-  zero_triangle <- matrix(
-    c(
-      80, 50, 25, 10,
-      100, 50, 30, 20,
-      90, 45, 25, 0,
-      80, 40, 0, 0,
-      70, 0, 0, 0
-    ),
-    nrow = 5,
-    byrow = TRUE
+
+  expect_equal(delay_pmf, complete_pmf, tolerance = 0.001)
+})
+
+test_that(
+  "get_delay_estimate works with every other day reporting of daily data",
+  {
+    sim_delay_pmf <- c(0.1, 0.2, 0.3, 0.1, 0.1, 0.1, 0.1)
+
+    counts <- c(30, 40, 50, 60, 70)
+
+    complete_triangle <- lapply(counts, function(x) x * sim_delay_pmf)
+    complete_triangle <- do.call(rbind, complete_triangle)
+
+    reporting_triangle <- replace_lower_right_with_NA(
+      complete_triangle,
+      structure = 2
+    )
+
+    # Get delay estimate
+    delay_pmf <- get_delay_estimate(
+      reporting_triangle = reporting_triangle
+    )
+    # Test that the function returns the expected PMF
+    expect_identical(delay_pmf, sim_delay_pmf)
+  }
+)
+
+test_that("get_delay_estimate handles complete reporting triangles", {
+  # Create a triangle with known delay PMF
+  partial_pmf <- c(0.4, 0.2, 0.2, 0.2)
+
+  # Generate counts for each reference date
+  partial_counts <- c(80, 100, 90, 80, 70)
+
+  # Create a complete triangle based on the known delay PMF
+  partial_complete <- lapply(partial_counts, function(x) x * partial_pmf)
+  partial_complete <- do.call(rbind, partial_complete)
+
+  # Create a reporting triangle with NAs in the lower right
+  partial_triangle <- replace_lower_right_with_NA(
+    partial_complete,
+    structure = 1
   )
 
-  expect_warning(get_delay_estimate(
-    reporting_triangle = zero_triangle,
+  delay_pmf <- get_delay_estimate(
+    reporting_triangle = partial_triangle,
     max_delay = 3,
-    n = 5
-  ))
-
-  # Test 11: zeros not just on bottom right
-  zero_triangle2 <- matrix(
-    c(
-      80, 50, 25, 10,
-      100, 0, 30, 20,
-      90, 45, 25, 0,
-      80, 40, 0, 0,
-      70, 0, 0, 0
-    ),
-    nrow = 5,
-    byrow = TRUE
+    n = 4
   )
 
-  expect_warning(get_delay_estimate(
-    reporting_triangle = zero_triangle2,
-    max_delay = 3,
-    n = 5
-  ))
+  expect_is(delay_pmf, "numeric")
+  expect_equal(sum(delay_pmf), 1, tolerance = 1e-6)
+  expect_equal(delay_pmf, partial_pmf, tolerance = 1e-6)
```

Comment:
Should we be adding tests for `.fill_triangle()` or are these tests of the wrapper fine?

Reply:
I would think it would be good to test that one in particular.

Reply:
I think we can though I don't think we need to address it in this PR as the functionality should be covered that is being added here. I think the tests as a whole need some work.